### PR TITLE
Support hashed string values in ini properties

### LINKED ISSUE #121: Support #enclosed# string values in ini files
```
[Section]
keyA = #valueA#   # Some comment
keyB = valueB#Some comment
```

### PR DESCRIPTION
**Pull request by arthurvd**
Fixes #121

### CODE REVIEW

File: hydrolib/core/io/ini/parser.py [hunk 1]
```diff
@@ -239,8 +239,26 @@ def _handle_datarow(self, line: str) -> None:
 
     def _retrieve_property_comment(self, line: str) -> Tuple[Optional[str], str]:
         if self._config.parse_comments and self._config.comment_delimiter in line:
-            key_value, comment = line.split(self._config.comment_delimiter, 1)
-            return comment.strip(), key_value.strip()
+            # key_value, comment = line.split(self._config.comment_delimiter, 1)
```
Comment (evetion) [inline hydrolib/core/io/ini/parser.py:242]: Remove comment

File: hydrolib/core/io/ini/parser.py [hunk 2]
```diff
@@ -239,8 +239,26 @@ def _handle_datarow(self, line: str) -> None:
...
+            parts = line.strip().split(self._config.comment_delimiter)
+            numhash = len(parts) - 1
```
Comment (evetion) [inline hydrolib/core/io/ini/parser.py:244]: Might be better to just use `line.count("#")`, and only split/strip when really required.

File: hydrolib/core/io/ini/parser.py [hunk 3]
```diff
@@ -239,8 +239,26 @@ def _handle_datarow(self, line: str) -> None:
...
+            if numhash == 1:
+                # normal value, simple comment: "key =  somevalue # and a comment "
+                return parts[-1].strip(), parts[0].strip()
+            elif parts[0] == "":
```
Comment (evetion) [inline hydrolib/core/io/ini/parser.py:248]: This logic is odd to read. Might be better to write `numhash == 2` (case without comment).

File: hydrolib/core/io/ini/parser.py [hunk 4]
```diff
@@ -239,8 +239,26 @@ def _handle_datarow(self, line: str) -> None:
...
+                # hashed value, possible with comment: "key = #somevalue# ..."
+                return (
+                    self._config.comment_delimiter.join(parts[3:]).strip()
+                    if numhash >= 3
+                    else None,
+                    self._config.comment_delimiter.join(parts[0:3]).strip(),
+                )
+            else:
```
Comment (evetion) [inline hydrolib/core/io/ini/parser.py:256]: With the other comments, this automatically becomes 3 or more hashes. So a hashed value and a comment (which may or not contain more hashes).

**Review by evetion — CHANGES_REQUESTED** [review]
Maybe also add a docstring/larger commented part that details the three cases you can have in terms of comments (the why of this function) together, not just in between the lines.

File: hydrolib/core/io/ini/parser.py (hunk 3 above)
Comment (arthurvd) [reply hydrolib/core/io/ini/parser.py:248]: It is deliberate: it catches these cases:
```
key = #valueA# # and a comment
key = #valueA with spaces # # Comment: and give me your #1 reason for writing such a comment?
```
if the first token of split is empty string (and numhas > 1), then we have a starting "#", which must always mean hash-string-quoting.

File: hydrolib/core/io/ini/parser.py (hunk 2 above)
Comment (arthurvd) [reply hydrolib/core/io/ini/parser.py:244]: is already done by existing if in line above: `[..]and self._config.comment_delimiter in line:`

File: hydrolib/core/io/ini/parser.py (hunk 3 above)
Comment (arthurvd) [reply hydrolib/core/io/ini/parser.py:248]: It catches this silly case:
```
key = valueA # and here is hash #2, funny huh?
```

File: hydrolib/core/io/ini/parser.py (hunk 4 above)
Comment (arthurvd) [reply hydrolib/core/io/ini/parser.py:256]: Please check whether with my explanations above any of this changes are still needed.

**Comment by arthurvd** [comment]
> Maybe also add a docstring/larger commented part that details the three cases you can have in terms of comments (the why of this function) together, not just in between the lines.

See latest commit 8d33250.

File: hydrolib/core/io/ini/parser.py (hunk 3 above)
Comment (arthurvd) [reply hydrolib/core/io/ini/parser.py]: See latest commit

**Pull request merged**